contents/scala/Scala.md

```diff
@@ -0,0 +1,241 @@
+<frontmatter>
+  title: Introduction to Scala
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Scala
+
+Author: [Wang Chao](https://github.com/fzdy1914)
+
+**Table of Contents**
+
+* [Overview](#overview)
+  * [Characteristic of Scala](#characteristic-of-scala)
+    * [Statically Typed](#statically-typed)
+    * [Object Oriented](#object-oriented)
+    * [Functional](#functional)
+* [Getting Started](#getting-started)
+* [Why Scala](#why-scala)
+  * [Operator Overloading](#operator-overloading)
+  * [Mixed-In Trait](#mixed-in-trait)
+  * [Type Enrichment](#type-enrichment)
+  * [Pattern Matching](#pattern-matching)
+
+# Overview
+
+**Scala** is a modern *multi-paradigm* programming language. It designed to integrate features of 
+*object-oriented* and *functional* languages in a *concise*, *elegant*, and *type-safe* way.
+
+## Characteristic of Scala
+
+### Statically Typed
+Scala is **statically typed**, which means it enforces type checking ike verifying and enforcing the constraints of 
+types at compile-time. 
+
+Scala allows [**type inference**](https://docs.scala-lang.org/tour/type-inference.html), which means it detects the data 
+type of an expression automatically. User is not required to annotate redundant type information in Scala. 
+
+In combination, these features provide a clean but reliable programming basis for the user.
+
+### Object Oriented
+Scala is a **pure OOP language**, which means every value in it is an object, including functions and primitives.
+
+Types and behavior of objects are described by *classes* and *traits*. 
```

Comment:
Provide a link to traits as not everyone knows what it is? Or can mention it is similar to java interface?